features/manage-position/Redeem.tsx

```diff
@@ -120,7 +120,9 @@ const Redeem = () => {
     };
 
     const setTokensToRedeemToMax = () => {
-      if (tokenBalance > posTokens) {
+      // `tokenBalance` and `posTokens` might be incorrectly rounded,
+      // so we round them arbitrarily to the same precision to compare.
+      if (tokenBalance.toFixed(4) >= posTokens.toFixed(4)) {
```

Comment:
how come you dont use the toWei, parseFixed or BN to compare these? it seems like this could cause an edge case but is ok in most cases. Theres some other parts where balances are cast with Number as well, im surprised this doesnt cause more problems.

Reply:
Yeah problem is that `tokenBalance` is passed as a `number` from the Position container. This is non-ideal and all the containers should ideally be passing only BN's to the front-end components.

Reply:
i think there are other issues with doing comparisons between numbers by converting them to strings first. This seems to work in many cases but here is one that fails:
`9.99.toFixed(4) > 10.0.toFixed(4)  == true`

Its because its comparing the values lexicographically since toFixed returns a string

Reply:
hmm I agree. So instead we could add a `balanceBN` to the Token container and use the BN value of the token balance instead of the truncated number

Reply:
@daywiss I fixed this using the solution above, please check